trainer.py

```diff
@@ -438,13 +440,115 @@ def _train(self):
+                 loss='sparse_categorical_crossentropy',
+                 metrics=None,
+                 layers_to_freeze=None,
+                 layers_to_copy=None,
+                 weight_map=None,
+                 **kwargs):
+        """
+        :param dataset: Keras dataset
+        :param n_classes: Number of output classes
+        :param epochs: Number of train Epochs
+        :param batch_size: Size of train batches
+        :param optimizer: Optimizer used for training
+        :param loss: Type of loss to use
+        :param metrics: Metrics for evaluation : List
+        :param layers_to_freeze: Optional list of layer indexes to set to
+            'untrainable', i.e. their weights cannot change during
+            training.
+        :param weight_map: Optional. Maps layer indexes to initial weight
+            values to use (instead of random init). Intended for use
+            with frozen layers.
+        """
+        super().__init__(dataset=dataset,
+                         n_layers=None,
+                         batch_size=batch_size,
+                         epochs=epochs,
+                         normalize_data=False,
+                         **kwargs)
+
+        self._n_classes = n_classes
+        self._optimizer = optimizer
+        self._loss = loss
+        self._metrics = metrics if metrics else ['accuracy']
+        self._layers_to_freeze = layers_to_freeze if layers_to_freeze else []
+        self._weight_map = weight_map if weight_map else {}
+        weighted_layers = self.get_weighted_layers_indices()
+        self._layers_to_freeze = [weighted_layers[i] for i in layers_to_freeze] if layers_to_freeze else []
+        self._weight_map = weight_map if weight_map else {}
+        if layers_to_copy and weight_map:
+            layers_to_copy = [weighted_layers[i] for i in layers_to_copy]
+            self._weight_map = {idx: weight_map[idx] for idx in layers_to_copy}
+
+    def create_model(self):
+        """
+        Based on https://keras.io/examples/cifar10_cnn/
+        :return: alexnet model
+        """
+        model = Sequential()
+        model.add(Conv2D(32, (3, 3), padding='same',
+                         input_shape=self._shape))
+        model.add(Activation('relu'))
+        model.add(Conv2D(32, (3, 3)))
+        model.add(Activation('relu'))
+        model.add(MaxPooling2D(pool_size=(2, 2)))
+        model.add(Dropout(0.25))
+
+        model.add(Conv2D(64, (3, 3), padding='same'))
+        model.add(Activation('relu'))
+        model.add(Conv2D(64, (3, 3)))
+        model.add(Activation('relu'))
+        model.add(MaxPooling2D(pool_size=(2, 2)))
+        model.add(Dropout(0.25))
+
+        model.add(Flatten())
+        model.add(Dense(512))
+        model.add(Activation('relu'))
+        model.add(Dropout(0.5))
+        model.add(Dense(self._n_classes))
+        model.add(Activation('softmax'))
+
+        return model
+
+    @classmethod
+    def get_weighted_layers_indices(cls):
```

Comment:
In `BaseTrainer` this method isn't a class method, Does this "override" work as expected?